File: 0401_0500/0427/0427.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    p = [int(x) for x in f.read().split()[1:]]
    p.sort()
    s = 0
    for x in p:
        if x - 1 <= s:
            s += x
        else:
            break
    q.write(str(s + 1))

File: 0401_0500/0427/0427.c
#include <stdio.h>
#include <stdlib.h>

int cmp(const void *a, const void *b)
{
	return *(int*)a - *(int*)b;
}

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int n, *p;
	long long s = 0;

	fscanf(f, "%d", &n);
	p = (int*)malloc(sizeof(int) * n);
	for (int i = 0; i < n; ++i)
		fscanf(f, "%d", p + i);
	qsort(p, n, sizeof(int), cmp);
	for (int i = 0; i < n; ++i)
		if (p[i] - 1 <= s)
			s += p[i];
		else
			break ;
	fprintf(q, "%I64d", s + 1);
	return 0;
}
